src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,38 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String brand;
+    private String model;
+    private String version;
+    private int numberOfDoors;
+    private Engine engine;
 
+    public Car(String brand, String model, String version, int numberOfDoors, Engine engine) {
+        this.brand = brand;
+        this.model = model;
+        this.version = version;
+        this.numberOfDoors = numberOfDoors;
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        return (Car) super.clone();
```

Comment:
please return to your previous implementation here
you need to create a new Car here and return it
you can use constructor, but you should pass not the same Engine object - but its clone

Reply:
I have made changes but still don't get what is the point to do like this? Do we want CarClone (during creation) to have different Engines (references to different objects) in both cars? Then I didn't achieve it and don't understand how to do it:(.